### PR TITLE
BUILD(client): Fix various issues

### LINKED ISSUE #5606: error: no match for ‘operator==’
### Description

Compilation fails with:
```cpp
[ 78%] Building CXX object src/mumble/CMakeFiles/mumble.dir/Markdown.cpp.o
/compile/mumble/mumble-1.4.230.src/src/mumble/Markdown.cpp: In function ‘void Markdown::escapeCharacter(QString&, int&)’:
/compile/mumble/mumble-1.4.230.src/src/mumble/Markdown.cpp:346:29: error: no match for ‘operator==’ (operand types are ‘QString’ and ‘QCharRef’)
  346 |  if (tmp.size() == 1 && tmp == str[offset]) {
      |                         ~~~ ^~ ~~~~~~~~~~~
      |                         |                |
      |                         QString          QCharRef
In file included from /usr/include/x86_64-linux-gnu/qt5/QtCore/QString:1,
                 from /compile/mumble/mumble-1.4.230.src/src/mumble/Markdown.h:9,
                 from /compile/mumble/mumble-1.4.230.src/src/mumble/Markdown.cpp:6:
/usr/include/x86_64-linux-gnu/qt5/QtCore/qstring.h:631:10: note: candidate: ‘bool QString::operator==(QLatin1String) const’
  631 |     bool operator==(QLatin1String s) const;
      |          ^~~~~~~~
/usr/include/x86_64-linux-gnu/qt5/QtCore/qstring.h:631:35: note:   no known conversion for argument 1 from ‘QCharRef’ to ‘QLatin1String’
  631 |     bool operator==(QLatin1String s) const;
      |                     ~~~~~~~~~~~~~~^
```

### Steps to reproduce

1. unpack mumble-1.4.230.tar.gz
2. cd to mumble-1.4.230.src
3. mkdir build; cd build
4. cmake ..
5. make

### Mumble version

1.4.230

### Mumble component

Both

### OS

Linux

### Reproducible?

Yes

### Additional information

<details>
<summary></summary>

```
-- The C compiler identification is GNU 9.4.0
-- The CXX compiler identification is GNU 9.4.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test COMPILER_HAS_MBIG_OBJ
-- Performing Test COMPILER_HAS_MBIG_OBJ - Failed
-- ##################################################
-- Mumble version:          1.4.0
-- Architecture:            x64
-- Build type:              Release
-- ##################################################
-- Qt5 component found: Core | Version: 5.5.1
-- Qt5 component found: Network | Version: 5.5.1
-- Qt5 component found: Xml | Version: 5.5.1
-- OpenSSL component found: Crypto | Version: 1.1.1
-- OpenSSL component found: SSL | Version: 1.1.1
-- Protobuf found | Version: 3.12.4
-- Qt5 component found: Concurrent | Version: 5.5.1
-- Qt5 component found: Sql | Version: 5.5.1
-- Qt5 component found: Svg | Version: 5.5.1
-- Qt5 component found: Widgets | Version: 5.5.1
-- Poco component not found: Zip
-- Regular Poco search failed - looking for Poco include dir manually...
-- Found Poco include dir at "/usr/include"
-- Found Poco Zip library at "/usr/lib/libPocoZip.so"
-- sndfile found | Version: 1.0.25
-- X11 component found: Xext
-- X11 component found: Xi
-- Found Git: /usr/bin/git (found version "2.35.1") 
CMake Warning at 3rdparty/opus/cmake/OpusPackageVersion.cmake:61 (message):
  Could not get package version.
Call Stack (most recent call first):
  3rdparty/opus/CMakeLists.txt:5 (get_package_version)


-- Opus project version: 0
-- Looking for floor in m
-- Looking for floor in m - found
-- Performing Test VLA_SUPPORTED
-- Performing Test VLA_SUPPORTED -- success
-- Looking for alloca.h
-- Looking for alloca.h - found
-- Looking for alloca
-- Looking for alloca - found
-- Looking for lrintf
-- Looking for lrintf - not found
-- Looking for lrint
-- Looking for lrint - not found
-- Looking for cpuid.h
-- Looking for cpuid.h - found
-- Check SIMD support by compiler
-- Looking for xmmintrin.h
-- Looking for xmmintrin.h - found
-- Performing Test SSE1_SUPPORTED
-- Performing Test SSE1_SUPPORTED - Success
-- Looking for emmintrin.h
-- Looking for emmintrin.h - found
-- Performing Test SSE2_SUPPORTED
-- Performing Test SSE2_SUPPORTED - Success
-- Looking for smmintrin.h
-- Looking for smmintrin.h - found
-- Performing Test SSE4_1_SUPPORTED
-- Performing Test SSE4_1_SUPPORTED - Success
-- Looking for immintrin.h
-- Looking for immintrin.h - found
-- Performing Test AVX_SUPPORTED
-- Performing Test AVX_SUPPORTED - Success
-- Performing Test FAST_MATH_SUPPORTED
-- Performing Test FAST_MATH_SUPPORTED - Success
-- Performing Test STACK_PROTECTOR_SUPPORTED
-- Performing Test STACK_PROTECTOR_SUPPORTED - Success
-- Performing Test HIDDEN_VISIBILITY_SUPPORTED
-- Performing Test HIDDEN_VISIBILITY_SUPPORTED - Success
-- Performing Test Wall_SUPPORTED
-- Performing Test Wall_SUPPORTED - Success
-- Performing Test W_SUPPORTED
-- Performing Test W_SUPPORTED - Success
-- Performing Test Wstrictprototypes_SUPPORTED
-- Performing Test Wstrictprototypes_SUPPORTED - Success
-- Performing Test Wextra_SUPPORTED
-- Performing Test Wextra_SUPPORTED - Success
-- Performing Test Wcastalign_SUPPORTED
-- Performing Test Wcastalign_SUPPORTED - Success
-- Performing Test Wnestedexterns_SUPPORTED
-- Performing Test Wnestedexterns_SUPPORTED - Success
-- Performing Test Wshadow_SUPPORTED
-- Performing Test Wshadow_SUPPORTED - Success
-- The following features have been enabled:

 * OPUS_BUILD_SHARED_LIBRARY, build shared library.
 * OPUS_ENABLE_FLOAT_API, compile with the floating point API (for machines with float library).
 * OPUS_HARDENING, run-time checks that are cheap and safe for use in production.
 * OPUS_INSTALL_PKG_CONFIG_MODULE, install pkg-config module.
 * OPUS_INSTALL_CMAKE_CONFIG_MODULE, install CMake package config module.
 * OPUS_VAR_ARRAYS, use variable length arrays for stack arrays.
 * OPUS_STACK_PROTECTOR, use stack protection.
 * OPUS_FORTIFY_SOURCE, add protection against buffer overflows.
 * OPUS_X86_MAY_HAVE_SSE, does runtime check for SSE1 support.
 * OPUS_X86_MAY_HAVE_SSE2, does runtime check for SSE2 support.
 * OPUS_X86_MAY_HAVE_SSE4_1, does runtime check for SSE4.1 support.
 * OPUS_X86_MAY_HAVE_AVX, does runtime check for AVX support.
 * OPUS_X86_PRESUME_SSE, assume target CPU has SSE1 support (override runtime check).
 * OPUS_X86_PRESUME_SSE2, assume target CPU has SSE2 support (override runtime check).

-- The following OPTIONAL packages have been found:

 * Git

-- The following features have been disabled:

 * OPUS_BUILD_TESTING, build tests.
 * OPUS_CUSTOM_MODES, enable non-Opus modes, e.g. 44.1 kHz & 2^n frames.
 * OPUS_BUILD_PROGRAMS, build programs.
 * OPUS_DISABLE_INTRINSICS, disable all intrinsics optimizations.
 * OPUS_FIXED_POINT, compile as fixed-point (for machines without a fast enough FPU).
 * OPUS_FLOAT_APPROX, enable floating point approximations (Ensure your platform supports IEEE 754 before enabling).
 * OPUS_ASSERTIONS, additional software error checking.
 * OPUS_FUZZING, causes the encoder to make random decisions (do not use in production).
 * OPUS_CHECK_ASM, enable bit-exactness checks between optimized and c implementations.
 * OPUS_FIXED_POINT_DEBUG, debug fixed-point implementation.
 * OPUS_USE_ALLOCA, use alloca for stack arrays (on non-C99 compilers).
 * OPUS_NONTHREADSAFE_PSEUDOSTACK, use a non threadsafe pseudostack when neither variable length arrays or alloca is supported.
 * OPUS_FAST_MATH, enable fast math (unsupported and discouraged use, as code is not well tested with this build option).
 * OPUS_X86_PRESUME_SSE4_1, assume target CPU has SSE4.1 support (override runtime check).
 * OPUS_X86_PRESUME_AVX, assume target CPU has AVX support (override runtime check).

-- Disabling warnings for targets: opus
INSTALL TARGETS - target opus has PUBLIC_HEADER files but no PUBLIC_HEADER DESTINATION.
-- Disabling warnings for targets: celt
-- Disabling warnings for targets: speex
-- Disabling warnings for targets: rnnoise
-- speech-dispatcher found | Version: 0.8.3
-- Qt5 component found: DBus | Version: 5.5.1
-- Qt5 component found: LinguistTools | Version: 5.5.1
-- ALSA found
-- Qt5 component found: Sql | Version: 5.5.1
-- File: /compile/mumble/build/src/murmur/murmur_autogen
-- Qt5 component found: DBus | Version: 5.5.1
-- Ice component found: Ice | Version: 3.5.1
-- Ice component found: IceSSL | Version: 3.5.1
-- Ice optional component found: IceUtil | Version: 3.5.1
-- Looking for sys/cdefs.h
-- Looking for sys/cdefs.h - found
-- 
If the 32 bit overlay library fails to compile, make sure the requirements are installed ("g++-multilib" package on Debian-based distributions).

-- Configuring done
-- Generating done
-- Build files have been written to: /compile/mumble/build
```
</details>

### Relevant log output

_No response_

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by Krzmbrzl**
See individual commits for details

Fixes #5606 

### Checks

- [x] My commits follow the [commit guidelines](https://github.com/mumble-voip/mumble/blob/master/COMMIT_GUIDELINES.md)

### CODE REVIEW

File: src/mumble/QtWidgetUtils.cpp [hunk 1]
```diff
@@ -11,6 +11,8 @@
 #include <QTextCursor>
 #include <QTextDocument>
 
+#include <cstdint>
```
Comment (davidebeatrici) [inline src/mumble/QtWidgetUtils.cpp:14]: I generally include standard headers at the top, followed by third-party ones (which would also match the change in `QtWidgetUtils.h`).

File: src/mumble/Markdown.cpp [hunk 2]
```diff
@@ -344,7 +344,7 @@ void escapeCharacter(QString &str, int &offset) {
 
 	tmp = tmp.toHtmlEscaped();
 
-	if (tmp.size() == 1 && tmp == str[offset]) {
+	if (tmp.size() == 1 && tmp[0] == str[offset]) {
```
Comment (davidebeatrici) [inline src/mumble/Markdown.cpp:347]:
```suggestion
	if (tmp.size() == 1 && tmp.front() == str[offset]) {
```

**Review by davidebeatrici — APPROVED** [review]

File: src/mumble/Markdown.cpp (hunk 2 above)
Comment (Krzmbrzl) [reply src/mumble/Markdown.cpp:347]: I think I'd prefer the more commonly known and recognized operator[] :thinking:

File: src/mumble/QtWidgetUtils.cpp (hunk 1 above)
Comment (davidebeatrici) [reply src/mumble/QtWidgetUtils.cpp:14]: I don't think we do.

**Comment by Krzmbrzl** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|1.4.x|[](https://github.com/mumble-voip/mumble/pull/5666)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**